### ISSUE TITLE
ENH: Alternative Kernels in HAC estimation

### ISSUE THREAD

**Issue by itsNH98** [statsmodels/statsmodels#8786]
Actually, RegressionResults 'kernel' only allows for Bartlett and Uniform kernel weighing functions in the context of robust linear regression, which are taken from "statsmodels.stats.sandwich_covariance.cov_hac". 

It's pretty clear from the literature (authors such as Kiefer, Voegelsang , Sun, Philips, etc.) that kernel selection has an important impact on results. 

Would it be complicated to include additional kernel functions, such as the ones that are present in MATLAB (https://www.mathworks.com/help/econ/hac.html) in statsmodels?

I am actually working on the subject so I wouldn't mind doing it.

**Comment by extraterrestrialchafik** [comment]
Hi! i find your issue really interesting i am a Data Scientist from France that would like to contribute to open source, what about we do it together? 

Which kernel functions do you think should be added?

**Comment by itsNH98** [comment]
I'd definitely be willing to do that (I'm actually from Montréal, so we have the language in common !) 

My background is in Financial Econometrics and Asset Pricing, so my ideas mostly come from this type of literature. A good recent paper on the subject is Sun & Yang (2020) "Testing-optimal kernel choice in HAR inference", which uses the Bartlett (already available), QS and Parzen for inference with fixed-b asymptotics. 

We could also think about adding the Tukey-Hanning and the Truncated kernels, amongst others. 

I would also be interested in providing documentation and guidelines for kernel and bandwidth choice, which are very hard to find in my experience. 

Tell me what you think !

**Comment by itsNH98** [comment]
@bashtage 

Yes, I have seen that you have numerous Kernels for LRV. I was thinking we could integrate those tools in StatsModels for completion and ease of use. 

Do you think doing a cleanup in the arch package could also be interesting? I see many Kernels have duplicate names, etc. 

Tell me what you think

**Comment by itsNH98** [comment]
@extraterrestrialchafik 

Yes, that seems to be the one.

As mentioned by @bashtage , we also have a lot of references in the arch package for this (see his previous comment) so that would come very helpful.

**Comment by extraterrestrialchafik** [comment]
Where do you want us to do it? Discord?

**Comment by bashtage** [comment]
@itsNH98 I don't think there are any duplicates there, but I'm happy to accept clean up PRs

**Comment by itsNH98** [comment]
@extraterrestrialchafik Add me NH98#7429
